src/components/Login.vue

```diff
@@ -134,15 +134,23 @@
   </div>
 </template>
 
-<script>
-import firebase from 'firebase/app';
+<script lang="ts">
+import Vue from 'vue';
+import firebase, { User } from 'firebase/app';
 
-const fb = require('../firebaseConfig.js');
+import * as fb from '@/firebaseConfig.ts';
```

Comment:
Now we must remove `.js` because it's no longer a js file.

Reply:
O I mean the difference between `import * as fb` and `const fb =`

Reply:
Usually on the web people use the `import` syntax. The `require` is only necessary on node, because node.js invented a module system before the ESModule (which uses `import`) becomes standard.
You might wonder for those typescript files, even if some of them runs on node, still use `import`  syntax. That's because typescript is a compiler, and it can transform it to use `require` after compilation.
In general, use `import` if you can.

Reply:
Gotcha. Thanks for clearing that up for me